feat(sdk): add kmp platform (#6452)
getsentry/sentry-docs: src/platform-includes/getting-started-install/kotlin-multiplatform.mdx

### Prerequisites (Android)

> Android require disabling auto-init on Android to not clash with the ContentProvider, which auto-initializes the Sentry Android SDK. To do so, add the following to the AndroidManifest.xml file under your androidMain source set:
> 

Reviewer: ```suggestion
Android requires disabling auto-init to not clash with the ContentProvider, which auto-initializes the Sentry Android SDK. To do so, add the following to the AndroidManifest.xml file under your androidMain source set:

```

> ```kotlin

Reviewer 2: ```suggestion
```xml
```

> ```kotlin
> <application>
>     <meta-data android:name="io.sentry.auto-init" android:value="false" />
> </application>
> ```

Reviewer 2: ```suggestion
```xml
<application>
    <meta-data android:name="io.sentry.auto-init" android:value="false" />
</application>
```
```

> To install the Kotlin Multiplatform SDK, you need to add the following to your `build.gradle.kts` file in your shared module:

Reviewer 2: Usually, there's multiple build.gradle.kts files, is this the root one or the app/build.gradle.kts one? let's be specific here.

Author: `shared module` should be the identifying folder here. When you create a brand new KMP app on Android studio or Intellij it will create a subproject `shared`, that's where the build.gradle is

Author: so maybe the file name should be shared/build.gradle.kts?

Reviewer 2: yeah thats a good idea

```kotlin {filename:build.gradle.kt}
repositories {
    mavenCentral()
}

kotlin {
  val commonMain by getting {
    dependencies {

>       api("io.sentry:sentry-kotlin-multiplatform:{{ packages.version('sentry.kotlin.kmp', '0.0.1') }}")

Reviewer 2: That requires https://github.com/getsentry/sentry-release-registry but it'll fallback to `0.0.1` so lets just be sure to fix the fallback version after being published.

>       api("io.sentry:sentry-kotlin-multiplatform:{{ packages.version('sentry.kotlin.kmp', '0.0.1') }}")

Reviewer 2: ```suggestion
      api("io.sentry:sentry-kotlin-multiplatform:{{ packages.version('sentry.kotlin.kmp', '0.0.1-alpha.2') }}")
```

    }
  }

  // Android target
  val androidMain by getting {
    dependsOn(commonMain)
  }

  // Apple targets:
  val iosMain by getting {
    dependsOn(commonMain)
  }

  cocoapods {
    summary = "Some description for the Shared Module"
    homepage = "Link to the Shared Module homepage"
    ios.deploymentTarget = "14.1"
    podfile = project.file("../iosApp/Podfile")

>     pod("Sentry", "~> {{ packages.version('sentry.cocoa', '8.2.0') }}")

Reviewer 2: Similar problem https://github.com/getsentry/sentry-kotlin-multiplatform/pull/56#discussion_r1136869751

>     pod("Sentry", "~> {{ packages.version('sentry.cocoa', '8.2.0') }}")

Author: Hm, currently there's no ideal solution so I agree syncing the cocoa sdk version manually is the way to go for now.

    framework {
      baseName = "shared"

>       export("io.sentry:sentry-kotlin-multiplatform:{{ packages.version('sentry.kotlin.kmp', '0.0.1') }}")

Reviewer 2: ```suggestion
      export("io.sentry:sentry-kotlin-multiplatform:{{ packages.version('sentry.kotlin.kmp', '0.0.1-alpha.2') }}")
```

    }
  }
  ```

> }

Reviewer: ```suggestion

```
